### PR TITLE
GlobalEfficiency Documentation

### PR DESCRIPTION
**Pull request by egolol**
#183 changed name. added documentation

### CODE REVIEW

File: braph/graph/measures/GlobalEfficiency.m [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            % initializes the property settings with SETTINGS. 
+            % Currently GLOBALEFFICIENCY does not utilizes SETTINGS.
+            %   
+            % See also Measure, Graph, Strength, Distance, GlobalEfficiency. 
+            
+            m = m@Measure(g, varargin{:});
+        end
+    end
+    methods (Access = protected)
+        function global_efficency = calculate(m)
```
Comment (maverickyuwei) [inline braph/graph/measures/GlobalEfficiency.m]: same change here effic"i"ency :P

File: braph/graph/measures/GlobalEfficiency.m [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            % CALCULATE calculates the globalefficiency value of a node
+            %
+            % GLOBALEFFICIENCY = CALCULATE(M) returns the value of the global_efficency of a
```
Comment (maverickyuwei) [inline braph/graph/measures/GlobalEfficiency.m]: effic"i"ency :P

File: braph/graph/measures/GlobalEfficiency.m [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+           
+            if g.is_measure_calculated('Distance')
+                D = g.getMeasure('Distance').getValue();
+            else
+                D = Distance(g, g.getSettings()).getValue();
+            end
+            
+            Di = D.^-1;  % inverse distance
+            Di(1:N+1:end) = 0;            
+            global_efficency = (sum(Di, 2) / (N-1));
```
Comment (maverickyuwei) [inline braph/graph/measures/GlobalEfficiency.m]: same change here effic"i"ency :P

File: braph/graph/measures/InGlobalEfficiency.m [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            % See also Measure, Graph, Strength, Distance, InGlobalEfficiency. 
+            
+            m = m@Measure(g, varargin{:});
+        end
+    end
+    methods (Access = protected)
+        function in_global_efficiency = calculate(m)
+            % CALCULATE calculates the in-globalefficiency value of a node
+            %
+            % INGLOBALEFFICIENCY = CALCULATE(M) returns the value of the in_global_efficency of a
```
Comment (maverickyuwei) [inline braph/graph/measures/InGlobalEfficiency.m]: same change here in_global_effic"i"ency :P

File: braph/graph/measures/OutGlobalEfficiency.m [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            % See also Measure, Graph, Strength, Distance, InGlobalEfficiency. 
+            m = m@Measure(g, varargin{:});
+        end
+    end
+    methods (Access = protected)
+        function out_global_efficiency = calculate(m)
+            % CALCULATE calculates the out-globalefficiency value of a node
+            %
+            % OUTGLOBALEFFICIENCY = CALCULATE(M) returns the value of the
+            % out_global_efficency of a node.
```
Comment (maverickyuwei) [inline braph/graph/measures/OutGlobalEfficiency.m]: same change here out_global_effic"i"ency :P

File: braph/graph/measures/OutGlobalEfficiency.m [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            % See also getName(), getDescription(). 
+            measure_class = 'OutGlobalEfficiency';
+        end
+        function name = getName()
+            % GETNAME returns the measure name
+            %
+            % NAME = GETNAME() returns the name of the out-globalefficiency measure.
+            %
+            % See also getClass(), getDescription(). 
+            name = 'Global Out Efficency';
```
Comment (maverickyuwei) [inline braph/graph/measures/OutGlobalEfficiency.m]: same change here Global Out Effic"i"ency :P

File: braph/graph/measures/GlobalEfficiency.m [hunk 7]
```diff
@@ -0,0 +1,145 @@
+classdef GlobalEfficiency < Measure
+    % GlobalEfficiency < Measure: GlobalEfficiency measure
+    % GlobalEfficiency provides the globalefficiency of a node for binary undirected (BU) and
```
Comment (annacanal) [inline braph/graph/measures/GlobalEfficiency.m]: global efficiency, separated in line 3

File: braph/graph/measures/GlobalEfficiency.m [hunk 8]
```diff
@@ -0,0 +1,145 @@
+classdef GlobalEfficiency < Measure
+    % GlobalEfficiency < Measure: GlobalEfficiency measure
+    % GlobalEfficiency provides the globalefficiency of a node for binary undirected (BU) and 
+    % weighted undirected (WU) graphs. It is calculated as average inverse
+    % shortest path length in the graph. 
+    % 
+    % GlobalEfficiency methods:
+    %   GlobalEfficiency                      - constructor with Measure properties.
```
Comment (annacanal) [inline braph/graph/measures/GlobalEfficiency.m]: this indentation is larger than the rest
Same in InGlobalEfficiency and OutGlobalEfficiency

File: braph/graph/measures/GlobalEfficiency.m [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+    %   getName                     - returns the name of globalefficiency measure.
+    %   getDescription              - returns the description of globalefficiency measure.
+    %   is_global                   - boolean, checks if globalefficiency measure is global.
+    %   is_nodal                    - boolean, checks if globalefficiency measure is nodal.
+    %   is_binodal                  - boolean, checks if globalefficiency measure if binodal.
+    %   getMeasure                  - returns the globalefficiency class.
+    %   getCompatibleGraphList      - returns a list of compatible graphs.
+    %   getCompatibleGraphNumber    - returns the number of compatible graphs.
+    %
+    % See also Measure, Graph, Strength, Distance, Degree.
```
Comment (annacanal) [inline braph/graph/measures/GlobalEfficiency.m]: Add some of the other efficiency related measures:
See also Measure, Graph, **InGlobalEfficiency, OutGlobalEfficiency, LocalEfficiency,**
Similarly in InGlobalEfficiency and OutGlobalEfficiency

File: braph/graph/measures/GlobalEfficiency.m [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+    methods
+        function m = GlobalEfficiency(g, varargin)
+            % GLOBALEFFICIENCY(G) creates globalefficiency with default measure properties.
+            % G is a graph (e.g, an instance of GraphBU, Graph WU). 
+            %   
+            % GLOBALEFFICIENCY(G, 'Settings', SETTINGS) creates globalefficiency measure and
+            % initializes the property settings with SETTINGS. 
+            % Currently GLOBALEFFICIENCY does not utilizes SETTINGS.
```
Comment (annacanal) [inline braph/graph/measures/GlobalEfficiency.m]: does not **utilize**
Same in InGlobalEfficiency and OutGlobalEfficiency

File: braph/graph/measures/GlobalEfficiency.m [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+            %   
+            % See also Measure, Graph, Strength, Distance, GlobalEfficiency. 
+            
+            m = m@Measure(g, varargin{:});
+        end
+    end
+    methods (Access = protected)
+        function global_efficiency = calculate(m)
+            % CALCULATE calculates the globalefficiency value of a node
```
Comment (annacanal) [inline braph/graph/measures/GlobalEfficiency.m]: global efficiency
Same in InGlobalEfficiency (in global efficiency) and OutGlobalEfficiency (out global efficiency)

File: braph/graph/measures/GlobalEfficiency.m [hunk 12]
```diff
@@ -1,11 +1,48 @@
...
     methods
         function m = GlobalEfficiency(g, varargin)
+            % GLOBALEFFICIENCY(G) creates globalefficiency with default measure properties.
+            % G is a graph (e.g, an instance of GraphBU, Graph WU). 
+            %   
+            % GLOBALEFFICIENCY(G, 'Settings', SETTINGS) creates globalefficiency measure and
+            % initializes the property settings with SETTINGS. 
+            % Currently GLOBALEFFICIENCY does not utilize SETTINGS.
+            %   
+            % See also Measure, Graph, Strength, Distance, GlobalEfficiency.
```
Comment (annacanal) [inline braph/graph/measures/GlobalEfficiency.m:34]: Aren't we already at GlobalEfficiency?
Maybe you wanted to add Out- or In-

Same at the constructor of InGlobalEfficiency

File: braph/graph/measures/OutGlobalEfficiencyAv.m [hunk 13]
```diff
@@ -1,11 +1,48 @@
 classdef OutGlobalEfficiencyAv < OutGlobalEfficiency
+    % OutGlobalEfficiency < Measure: OutGlobalEfficiency measure
+    % OutGlobalEfficiency provides the out-global efficiency of a node for binary directed (BD) and
```
Comment (annacanal) [inline braph/graph/measures/OutGlobalEfficiencyAv.m:3]: OutGlobalEfficiencyAv, here and everywhere

File: braph/graph/measures/OutGlobalEfficiencyAv.m [hunk 14]
```diff
@@ -1,11 +1,48 @@
...
+    %   is_nodal                    - boolean, checks if out-globalefficiency measure is nodal.
+    %   is_binodal                  - boolean, checks if out-globalefficiency measure if binodal.
+    %   getMeasure                  - returns the out-globalefficiency class.
+    %   getCompatibleGraphList      - returns a list of compatible graphs.
+    %   getCompatibleGraphNumber    - returns the number of compatible graphs.
+    %
+    % See also Measure, Graph, Strength, Distance, Degree, InGlobalEfficiency, LocalEfficiency,
     methods
         function m = OutGlobalEfficiencyAv(g, varargin)
+            % OUTGLOBALEFFICIENCY(G) creates out-globalefficiency with default measure properties.
```
Comment (annacanal) [inline braph/graph/measures/OutGlobalEfficiencyAv.m:27]: OUTGLOBALEFFICIENCYAV(G)

File: braph/graph/measures/GlobalEfficiency.m (hunk 12 above)
Comment (egolol) [reply braph/graph/measures/GlobalEfficiency.m:34]: dont understand review

**Pull request merged**